### PR TITLE
move DataConverter to shared folder and add it to the Go docs

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

Go didn't have DataConverter docs (which i needed for a customer) so i'm reusing what the Java docs have

## Notes to reviewers

- Go Workflows doc now has an added DataConverter section https://deploy-preview-425--mystifying-fermi-1bc096.netlify.app/docs/go/workflows#custom-serialization-and-workflow-security which is now shared with Java https://deploy-preview-425--mystifying-fermi-1bc096.netlify.app/docs/java/workflows#workflow-method-arguments
- we also moved https://docs.temporal.io/docs/go/continue-as-new into https://deploy-preview-425--mystifying-fermi-1bc096.netlify.app/docs/go/workflows#large-event-histories (to align with Java's similar move)

the rest are minor moves and lint fixes

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -74,7 +75,11 @@
 
 [[redirects]]
   from = "/docs/go-continue-as-new"
-  to = "/docs/go/continue-as-new"
+  to = "/docs/go/workflows"
+
+[[redirects]]
+  from = "/docs/go/continue-as-new"
+  to = "/docs/go/workflows"
```
Comment (sw-yx) [inline netlify.toml:82]:
```suggestion
  to = "/docs/go/workflows#large-event-histories"

[[redirects]]
  from = "/docs/go/continue-as-new"
  to = "/docs/go/workflows#large-event-histories"
```

File: docs/shared/dataconverter.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+The default implementation uses JSON serializer, but you can use any alternative serialization mechanism.
+
+The values passed to Workflows through invocation parameters or returned through a result value are recorded in the execution history.
+
+Even though Workflow execution history is cached in the Workers, in the case of Worker failure, the full execution history has to be transferred from the Temporal service to the Workflow Workers.
+
+In those cases a large execution history could adversely impact the performance of your Workflow. Be mindful of the amount of data that you transfer via Activity invocation parameters or return values.
+Otherwise, no additional limitations exist on Activity implementations.
+
+We discuss how to work around the history size limitations with `ContinueAsNew` in the <a href={props.continueAsNewURL}>Large Event Histories</a> section.
```
Comment (sw-yx) [inline docs/shared/dataconverter.md:11]: note for @tsurdilo and @flossypurse - notice how i use `props.href` and `props.continueAsNewURL` - this is how to make shared content still reflect the individual SDK's links and other special needs.

so for example on the java side i can use `<DataConverter href="https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/common/converter/DataConverter.html" continueAsNewURL="#large-event-histories" />` and the Go side can have different props.

File: docs/shared/dataconverter.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Workflow method arguments and return values are serializable to a byte array using <a href={props.href}>the provided DataConverter interface</a>.
```
Comment (sw-yx) [inline docs/shared/dataconverter.md:1]:
```suggestion
Workflow method arguments and return values are serializable to a [Payload](https://github.com/temporalio/api/blob/4c2f6a281fa3fde8b0a24447de3e0d0f47d230b4/temporal/api/common/v1/message.proto#L49) protobuf that contains a bytearray as well as metadata map. You can use <a href={props.href}>the SDK's DataConverter interface</a> to do this.
```

File: docs/shared/dataconverter.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+The default implementation uses JSON serializer, but you can use any alternative serialization mechanism.
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/shared/dataconverter.md:2]: adding in the appropriate line

File: docs/java/workflows.md [hunk 5]
```diff
@@ -325,20 +313,20 @@ public class GreetingWorkflowImpl implements GreetingWorkflow {
...
+An asynchronous start initiates a Workflow execution and immediately returns to the caller.
 This is the most common way to start Workflows in a live environment.
 
 <!--SNIPSTART money-transfer-project-template-java-workflow-initiator-->
 <!--SNIPEND-->
 
-If you need to wait for the completion of a Workflow after an asynchronous start, the most straightforward way is to call the blocking Workflow instance again. 
-If `WorkflowOptions.WorkflowIdReusePolicy` is not set to `AllowDuplicate`, then instead of throwing `DuplicateWorkflowException`, it reconnects to an existing Workflow and waits for its completion. 
+If you need to wait for the completion of a Workflow after an asynchronous start, the most straightforward way is to call the blocking Workflow instance again.
+If `WorkflowOptions.WorkflowIdReusePolicy` is not set to `AllowDuplicate`, then instead of throwing `DuplicateWorkflowException`, it reconnects to an existing Workflow and waits for its completion.
```
[the remark this answers is not in the document]
Comment (tsurdilo) [reply docs/java/workflows.md]: added

**Review by flossypurse — APPROVED** [review]
Wicked!
Love how we are using Shared content here.

**Comment by sw-yx** [comment]
thanks for the help everyone

**Pull request merged**